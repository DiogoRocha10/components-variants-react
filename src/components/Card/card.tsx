import styles from "./card.module.css"
import cn from "clsx"

interface CardProps {
	title: string
	children: string
	className?: string
}

export const Card = ({ title, className, children }: CardProps) => {
	return (
		<div className={cn(className, styles["-card--container"])}>
			<div className={styles["-card--content"]}>
				<div className={styles["-card--header"]}>
					<h3 className={styles["-card--title"]}>{title}</h3>
				</div>
				<span className={styles["-card--body"]}>{children}</span>
			</div>
		</div>
	)
}
