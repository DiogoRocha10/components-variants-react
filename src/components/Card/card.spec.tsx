import { render } from "@testing-library/react"
import { Card } from "./card"
import renderer from "react-test-renderer"

it("text", () => {
	const { getByText } = render(<Card title="Card" children="Card content" />)
	expect(getByText("Card")).toBeTruthy()
})

it("renders correctly", () => {
	const tree = renderer
		.create(<Card title="Card" children="Card content" />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
