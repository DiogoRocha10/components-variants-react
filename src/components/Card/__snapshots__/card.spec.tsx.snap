// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="-card--container"
>
  <div
    className="-card--content"
  >
    <div
      className="-card--header"
    >
      <h3
        className="-card--title"
      >
        Card
      </h3>
    </div>
    <span
      className="-card--body"
    >
      Card content
    </span>
  </div>
</div>
`;
